contracts/SpokePool.sol

```diff
@@ -1049,16 +1055,18 @@ abstract contract SpokePool is
                 fillAmountPreFees,
                 relayData.realizedLpFeePct + relayExecution.updatedRelayerFeePct
             );
+        }
 
-            if (relayExecution.payoutAdjustmentPct != 0) {
-                // If payoutAdjustmentPct is positive, then the recipient will receive more than the amount they
-                // were originally expecting. If it is negative, then the recipient will receive less.
-                // -1e18 is -100%. Because we cannot pay out negative values, that is the minimum.
-                require(relayExecution.payoutAdjustmentPct >= -1e18, "payoutAdjustmentPct too small");
+        // This can only happen in a slow fill, where the contract is funding the relay, so the maxTokensToSend can be ignored.
+        // This is why we don't check that the amountToSend is less than or equal to maxTokensToSend.
+        if (relayExecution.payoutAdjustmentPct != 0) {
+            // If payoutAdjustmentPct is positive, then the recipient will receive more than the amount they
+            // were originally expecting. If it is negative, then the recipient will receive less.
+            // -1e18 is -100%. Because we cannot pay out negative values, that is the minimum.
+            require(relayExecution.payoutAdjustmentPct >= -1e18, "payoutAdjustmentPct too small");
 
-                // Note: since _computeAmountPostFees is typicaly intended for fees, the signage must be reversed.
-                amountToSend = _computeAmountPostFees(amountToSend, -relayExecution.payoutAdjustmentPct);
-            }
+            // Note: since _computeAmountPostFees is typicaly intended for fees, the signage must be reversed.
+            amountToSend = _computeAmountPostFees(amountToSend, -relayExecution.payoutAdjustmentPct);
```

Comment:
@nicholaspai technically payoutAdjustmentPct could be as large as 2^255, which would make the amountToSend extremely large. Do you think we should add any upper bounds in the contract to payoutAdjustment?

Alternatively, do you think we should ensure that the final amountToSend is <= the maxSendAmount passed into this function. Feels a bit odd that the contract allows the payout to be larger than maxSendAmount in the slow relay case (even though slow relays are trusted).

Thoughts?

Reply:
i think adding an upper limit to prevent math issues is definitely a good idea. I can't see any reason why payoutAdjustmentPct would even be greater than say 1000% for example. Even that is stretching it

We have limits on `realizedLpFeePct` and `relayerFeePct` in _fillRelay so why not also have one for this pct to give us peace of mind?

Reply:
The new check on payoutAdjustmnetPct LGTM :+1: